vkubelet/pod.go

```diff
@@ -21,6 +30,114 @@ func addPodAttributes(span *trace.Span, pod *corev1.Pod) {
 	)
 }
 
+func (s *Server) onAddPod(ctx context.Context, obj interface{}) {
+	ctx, span := trace.StartSpan(ctx, "onAddPod")
+	defer span.End()
+	logger := log.G(ctx).WithField("method", "onAddPod")
+
+	pod := obj.(*corev1.Pod)
+
+	if pod == nil {
+		span.SetStatus(trace.Status{Code: trace.StatusCodeInvalidArgument, Message: fmt.Sprintf("Unexpected object from event: %v", obj)})
+		logger.Errorf("obj is not a valid pod: %v", obj)
+		return
+	}
+
+	addPodAttributes(span, pod)
+
+	logger.Debugf("Receive added pod '%s/%s' ", pod.GetNamespace(), pod.GetName())
+
+	if s.resourceManager.UpdatePod(pod) {
+		span.Annotate(nil, "Add pod to synchronizer channel.")
+		s.podCh <- pod
+	}
+}
+
+func (s *Server) onUpdatePod(ctx context.Context, obj interface{}) {
+	ctx, span := trace.StartSpan(ctx, "onUpdatePod")
+	defer span.End()
+	logger := log.G(ctx).WithField("method", "onUpdatePod")
+
+	pod := obj.(*corev1.Pod)
+
+	if pod == nil {
+		span.SetStatus(trace.Status{Code: trace.StatusCodeInvalidArgument, Message: fmt.Sprintf("Unexpected object from event: %v", obj)})
+		logger.Errorf("obj is not a valid pod: %v", obj)
+		return
+	}
+
+	addPodAttributes(span, pod)
+
+	logger.Debugf("Receive updated pod '%s/%s'", pod.GetNamespace(), pod.GetName())
+
+	if s.resourceManager.UpdatePod(pod) {
+		span.Annotate(nil, "Add pod to synchronizer channel.")
+		s.podCh <- pod
+	}
+}
+
+func (s *Server) onDeletePod(ctx context.Context, obj interface{}) {
+	ctx, span := trace.StartSpan(ctx, "onDeletePod")
+	defer span.End()
+	logger := log.G(ctx).WithField("method", "onDeletePod")
+
+	pod := obj.(*corev1.Pod)
+
+	if pod == nil {
+		span.SetStatus(trace.Status{Code: trace.StatusCodeInvalidArgument, Message: fmt.Sprintf("Unexpected object from event: %v", obj)})
+		logger.Errorf("obj is not a valid pod: %v", obj)
+		return
+	}
+
+	addPodAttributes(span, pod)
+
+	logger.Debugf("Receive deleted pod '%s/%s'", pod.GetNamespace(), pod.GetName())
+
+	if s.resourceManager.DeletePod(pod) {
+		span.Annotate(nil, "Add pod to synchronizer channel.")
+		s.podCh <- pod
+	}
+}
+
+func (s *Server) startPodSynchronizer(ctx context.Context, id int) {
+	logger := log.G(ctx).WithField("method", "startPodSynchronizer").WithField("podSynchronizer", id)
+	logger.Debug("Start pod synchronizer")
+
+	for {
+		select {
```

Comment:
This can just be `for pod := range s.podCh`, which will break when the channel is closed.